lib/practice_exercises.rb

```diff
@@ -1,13 +1,32 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) - Need to go through each index of the array to compare its values.
+# Space Complexity: O(1)
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  list.each do |i|
+    if list[i] == list[i+1]
+      list.delete_at(i)
+      i += 1
+    end
+  end
+    return list
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# referred to solution from https://leetcode.com/problems/longest-common-prefix/discuss/291314/Ruby-Solution
+# and added another condtion '&& !(word[-1] == base[i])' to pass test: "will work for the strings with the common prefix in the rear"
+
+# Time Complexity: O(n)
+# Space Complexity: O(2)
 def longest_prefix(strings)
-  raise NotImplementedError, "Not implemented yet"
-end
+  return "" if strings.empty? 
 
+  result = ""
+  base = strings[0]
+  
+  base.each_char do |i|
+    if strings.all?{|word| word[i] == base[i] && !(word[-1] == base[i])}
```

Comment:
Remember that `strings.all` traverses the entire list of strings, so it runs in O(n), which again causes your method to be O(n<sup>2</sup>).

Otherwise this also works.